### ISSUE TITLE
Dev mode is not re building / reloading in monorepo context (nx)

### ISSUE THREAD

**Issue by npirotte** [remix-run/remix#2983]
### What version of Remix are you using?

1.4.1

### Steps to Reproduce

* Create a monorepo with NX (I assume it's the same with turbo repo)
* Create a remix application
* Create a library
* Import something from the library into the application
* Start the application in dev mode
* Save some change in the library file.

Alternative that might go quicker:
* Fork https://github.com/npirotte/hello-visx
* run `npm i`
* run `nx dev demo`
* do some changes in `packages/bar-chart/src/lib/bar-chart.tsx`

### Expected Behavior

App rebuild and reload.

### Actual Behavior

Nothing happens, I have to do some change in the app itself to have it reloading.

**Comment by npirotte** [comment]
After more testing, I understand that remix listen to events on the `appDirectory`, while my expectation are that it listen to `appDirectory` + dependencies changes (in monorepo context, most of the files will be out of appDirectory).

**Comment by npirotte** [comment]
As an easy and simple solution, I can propose to add an optional configuration in the remix config:

```ts
export interface AppConfig {
  /**
  * List of extra paths or globs to watch changes on in addition to the main app directory.
  * relative to `remix.config.js`. Defaults to `[]`.
  */
  watch?: string | string[]
}
```

This would the be merge to the current list of watched directories:

```
// compiler.ts
let extraWatch = Array.isArray(config.watch) ? config.watch : [config.watch];
let toWatch = [config.appDirectory, ...extraWatch];
if (config.serverEntryPoint) {
  toWatch.push(config.serverEntryPoint);
}
```

**Comment by npirotte** [comment]
Already created an example implementation here: https://github.com/remix-run/remix/pull/2984

**Comment by fknop** [comment]
Having the same issue with NX. Is there an eventual workaround until the fix lands?

**Comment by npirotte** [comment]
Didn't found anything yet, currently I simply keep the root.tsx file opened, and I hit save on that one when I want to rebuild the app.

**Comment by jdnichollsc** [comment]
@fknop maybe using [patch-package](https://www.npmjs.com/package/patch-package) with the changes created by @npirotte in that PR 🤔

**Comment by MichaelDeBoey** [comment]
Closed by #3188
